### PR TITLE
[SPARK-23586][SQL] Add interpreted execution to WrapOption

### PR DESCRIPTION
**Pull request by mgaido91**
## What changes were proposed in this pull request?

The PR adds interpreted execution to WrapOption.

## How was this patch tested?

added UT

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala [hunk 1]
```diff
@@ -75,4 +75,14 @@ class ObjectExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
       checkEvaluation(unwrapObject, expected, InternalRow.fromSeq(Seq(input)))
     }
   }
+
+  test("SPARK-23586: WrapOption should support interpreted execution") {
+    val cls = ObjectType(classOf[java.lang.Integer])
+    val inputObject = BoundReference(0, cls, nullable = true)
+    val unwrapObject = WrapOption(inputObject, cls)
```
Comment (hvanhovell) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala]: NIT: `unwrapObject`?

**Review by hvanhovell — APPROVED** [review]
LGTM - pending Jenkins. Can you also touch up the name in the test case?

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala (hunk 1 above)
Comment (mgaido91) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala]: sorry...

Comment (hvanhovell) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala]: no problem :)

**Comment by hvanhovell** [comment]
Merging to master. Thanks!

**Pull request closed without merging**